src/TraceEvent/Symbols/SymbolReader.cs

```diff
@@ -1901,11 +1912,21 @@ protected virtual string GetSourceFromSrcServer()
             if (url != null)
             {
                 HttpClient httpClient = new HttpClient();
+
+                var authorizationHeader = this._symbolModule.SymbolReader.AuthorizationHeaderForSourceLink;
+                if (authorizationHeader != null)
+                {
+                    httpClient.DefaultRequestHeaders.Add("Authorization", authorizationHeader);
+                }
+
                 HttpResponseMessage response = httpClient.GetAsync(url).Result;
 
                 response.EnsureSuccessStatusCode();
                 Stream content = response.Content.ReadAsStreamAsync().Result;
-                string cachedLocation = GetCachePathForUrl(url);
+                string cachedLocation = Path.Combine(
+                    _symbolModule.SymbolReader.SourceCacheDirectory,
+                    BitConverter.ToString(SHA256.Create().ComputeHash(Encoding.UTF8.GetBytes(url.ToUpperInvariant())))
```

Comment:
We should probably be caching the result of ```SHA256.Create``` as this is going to be called once per symbol file.